### DISCUSSION TITLE
Question about Adaptivity block setting in MOOSE

### DISCUSSION THREAD

**Discussion by abc-hy** [idaholab/moose#24657] [Q&A Modules: Phase field] [answered]
Hi everyone,

I was confused about the setting of adaptivity block in MOOSE. In the figure shown below, what does the coarsen factor mean? Does it mean, divided by 0.1 each time? 
So, for example, if the original grid size is 0.2, then, for this adaptivity block setting, the coarsen factor can be applied for 3 times (because max_h_level is 3), and each time is divided by 0.1? So first time is 0.2/0.1=2, second time is 2/0.1=20, third time is 20/0.1=200. Therefore, the maximum possible grid size we can get is 200? Am I understanding correctly?
![image](https://github.com/idaholab/moose/assets/63679626/5da7195e-7aac-4f2d-a77a-e7954f5dcdbb)

Thank you very much,
Best,

**Comment by GiudGiud** [comment, marked as answer]
Hello

This parameter is forwarded to a libmesh class for mesh refinement
https://mooseframework.inl.gov/docs/doxygen/libmesh/classlibMesh_1_1MeshRefinement.html#ac65900f46067a75f67814fc91ef05d4b

It means the fraction of elements to refine. 
See the class documentation:
`This method picks the top refine_fraction * n_elem elements for refinement and the bottom coarsen_fraction * n_elem elements for coarsening.`

Refinement is not done in arbitrary increments. A quad or a tri is split in 4 in each step at most. A hex is split in 8 hexes. We dont have any other divisions available (except from refining multiple times)

Guillaume

**Comment by abc-hy** [comment]
So, you mean that the grid size can only be decreased/refined, it cannot be
increased? So the maximum grid size is the original grid size you set, am i
right?

**Reply by GiudGiud** [reply]
We can only undo refinement to increase. So you have to start with a mesh as coarse as you ll ever want it to be.

**Comment by abc-hy** [comment]
Actually, i wanted to calculate the uncertainty of the model. So i need the
maximum possible discretization of x......
So the grid size will be divided by the coarsen factor to increase, is that
correct?

**Reply by GiudGiud** [reply]
No it's not correct.
The grid size will be divided by 4 (assuming quads) on every step of refinement, at the elements chosen for refinement (depending on the refine_fraction)

**Comment by abc-hy** [comment]
SO, grid size will not be increased? What's the use of coarsen_factor?

**Comment by abc-hy** [comment]
If grid size cannot be increased, why do we need coarsen factor there?

**Reply by GiudGiud** [reply]
Coarsening is used to undo levels of refinement.
You should just try it out on a square problem with dummy variables to see what it does
